mod/ckeditor/views/default/elgg/ckeditor.js

```diff
@@ -13,13 +13,19 @@
 define(function (require) {
 	var elgg = require('elgg');
 	require('elgg/init');
+
 	var $ = require('jquery');
 	require('jquery.ckeditor');
+
 	var CKEDITOR = require('ckeditor');
-	var config = require('elgg/ckeditor/config');
 
 	var elggCKEditor = {
-		bind: function (selector) {
+		bind: function (selector, config) {
+			if (config === undefined) {
+				config = require('elgg/ckeditor/config');
```

Comment:
You caught the wrong impression :) All appearances of `require('foo')` are pre-loaded before the module function is called. So here we're still loading it no matter what. Seeing how RequireJS uses [this pattern](https://github.com/requirejs/requirejs/blob/2.3.2/require.js#L16) will make this clearer.